docs/release-procedure.md

```diff
@@ -0,0 +1,209 @@
+prow configs, and more. All you need to do is gather a few credentials
+in order to use the tool.
+
+## Release Tool Credentials
+
+You must have 2 items before you can create a release.
+
+1. **GPG key and passphrase files used for creating signed releases.**
+
+[Instructions for adding GPG key to your github account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account)
+
+After adding the GPG key to github, export both the key and passphrase to files.
+Be aware that this results in the key and passphrase being placed into a plain
+text file on your machine. Make sure you don't place this in shared storage.
+
+**Example of exporting key to file**
+
+```gpg --export-secret-key -a <your-email-address> > ${HOME}/gpg-private```
+
+**Example of putting passphrase in file**
+
+```echo "<insert your passphrase here>" > ${HOME}/gpg-passphrase```
+
+2. **Github API token file used for accessing the github api**
+
+When you create this token the only permission you need to give it is the
+ability to access github repositories. That's it.
+
+[Instructions for creating access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+**Place your token in a text file such as** ```${HOME}/github-api-token```
+
+## Release Tool Usage Examples
+
+Once you have your credentials in files, the kubevirt release tool handles
+all the rest. All you need to do is provide your credentials and tell the tool
+what release you want to make.
+
+Place the paths to your credential files in the following environment variables.
+
+```
+export GPG_PRIVATE_KEY_FILE="${HOME}/gpg-private"
+export GPG_PASSPHRASE_FILE="${HOME}/gpg-passphrase"
+export GITHUB_API_TOKEN_FILE="${HOME}/github-api-token"
+```
+
+Now you can use the release tool to do whatever you'd like. Note that you can
+use the ```--dry-run=true``` argument to test a change before executing it.
+
+**Example: creating a new release branch with the initial release candidate v0.31.0-rc.0**
+```
+hack/release.sh --new-branch release-0.31 --new-release v0.31.0-rc.0 --dry-run=false
+```
+
+**Example: Creating a new rc v0.31.0-rc.0**
+```
+hack/release.sh --new-release v0.31.0-rc.0 --dry-run=false
+```
+
+**Example: Promoting a release candidate v0.31.0-rc-1 to official v0.30.0 release.**
+```
+hack/release.sh --promote-release-candidate v0.31.0-rc-1 --dry-run=false
+```
+
+**Example: Creating a patch release v0.31.1. The branch will automatically be detected.**
+```
+hack/release.sh --new-release v0.31.1 --dry-run=false
+```
+
+## Creating New Minor Releases
+The release process is mostly automatic and consists of the following steps:
+
+1. Create the branch and initial RC.
+
+   ```hack/release.sh --new-branch $RELEASE_BRANCH --new-tag ${TAG}.rc.0```
+
+2. Wait 5 business days
+
+3. Promote RC to official release if no blockers exist.
+
+   ```hack/release.sh --promote-release-candidate ${TAG}.rc.0```
```

Comment:
This is no longer accurate. Maybe we should replace items 2 and 3 here with

```
2. Continue to create RC's according to the release schedule

3. Once it is time for an official release, promote the most recent RC if no blockers exist.

```hack/release.sh --promote-release-candidate ${TAG}.rc.0```

```

Reply:
Fixed